itests/hive-unit/src/test/java/org/apache/hive/beeline/TestHplSqlViaBeeLine.java

```diff
@@ -610,6 +610,25 @@ public void testNullCast() throws Throwable {
     testScriptFile(SCRIPT_TEXT, args(), "^(.(?!(NullPointerException)))*$", OutStream.ERR);
   }
 
+  @Test
+  public void testTableAliasInColumnName() throws Throwable {
+    String SCRIPT_TEXT =
+        "DROP TABLE IF EXISTS input;\n" +
+        "DROP TABLE IF EXISTS result;\n" +
+        "CREATE TABLE input (col1 string, col2 int);\n" +
+        "CREATE TABLE result (res string);\n" +
+        "INSERT INTO input VALUES('Hive', 3);\n" +
+        "CREATE PROCEDURE p1() AS\n" +
+        "  BEGIN\n" +
+        "    FOR rec IN (select tab.col1, tab.col2 num from input tab) LOOP\n" +
```

Comment:
Will this fail if 2 columns have the same alias name defined and within the LOOP we try to reference it ?  It would be good to have a meaningful error such as 'ambiguous column reference'.   
If this is not supported in the current code, it is ok to file a separate Jira and go ahead with this.

Reply:
It is just giving 'Unhandled exception in HPL/SQL' and in HS2 log file also no useful information. I will raise a separate issue to handle it.